### ISSUE TITLE
What is the proper way to prepare a label .txt file for multi-label detection?

### ISSUE THREAD

**Issue by bhoo-git** [ultralytics/ultralytics#15758]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I am looking to prepare a dataset for multi-label detection, where each object in an image could belong to multiple classes. I understand that the .txt file for labels needs to have the following format for each object in an image: `class xc yc w h`. To suit this for a multi-label case, would it suffice to simply add duplicate bounding boxes with different classes? Or is there a way to specify multiple labels for each bounding box in the same line?

For example, given this image with three objects:
![Results-from-object-detection-and-tracking-The-car-and-the-people-framed-in-green-are](https://github.com/user-attachments/assets/1110b17a-14a1-4185-add2-b58ada947724)

 I want the car to have the labels [vehicle, suv], and the two people to have the labels [person, male]. Can I make the label.txt for this image look like either of the options below? Which one would work to train a multi-label detection model? I saw Option 2 in one of the responses from @glenn-jocher, but I believe this format does not actually work.

Option 1:
```txt
cls_1 xc_1 yc_1 w_1 h_1
cls_2 xc_1 yc_1 w_1 h_1 
cls_3 xc_2 yc_2 w_2 h_2
cls_4 xc_2 yc_2 w_2 h_2
cls_3 xc_3 yc_3 w_3 h_3
cls_4 xc_3 yc_3 w_3 h_3
```
Option 2:
```txt
cls_1 xc_1 yc_1 w_1 h_1 cls_2
cls_3 xc_2 yc_2 w_2 h_2 cls_4
cls_3 xc_3 yc_3 w_3 h_3 cls_4
```

Would greatly appreciate it if I could get some support beyond the auto-generated response. Thanks in advance for the help!

**Comment by Y-T-G** [comment]
The first method would treat each box as a separate ground truth box, and the second one doesn't work. I don't think the training allows for multi-label training as it is, at least not effectively. There are certain restrictions like, each anchor only being allowed to he assigned to a single ground truth box, so the first method will not work well:
https://github.com/ultralytics/ultralytics/blob/62094bd03fc6d14726a8362ccb7e2872fe6b7909/ultralytics/utils/tal.py#L247

You would have to make changes in the way the data is loaded such that you don't have to create duplicate boxes, but you still have multiple classes for the same box instead.

**Comment by KungFuPandaPro** [comment]
I have the same question?do you deal with it?

**Comment by glenn-jocher** [comment]
@KungFuPandaPro for multi-label detection, YOLO currently does not support multiple labels for a single bounding box directly. Each bounding box should have a single class label. If you need multi-label detection, consider modifying the data loading and training process to handle multiple classes per bounding box.

**Comment by ZJZ0405** [comment]
I think the preset network structure of yolov8 does not support multiple classes, you may need to modify the detection header

**Comment by KungFuPandaPro** [comment]
> I think the preset network structure of yolov8 does not support multiple classes, you may need to modify the detection header

网络头可以不改吧，就是标签改一下，主要改损失和dataloder

**Comment by KungFuPandaPro** [comment]
> I think the preset network structure of yolov8 does not support multiple classes, you may need to modify the detection header

如果你也有这样的需求，我们一起讨论开发一下

**Comment by ZJZ0405** [comment]
但yolov8这样的解耦可以对多标签用不同的结构吧

**Comment by glenn-jocher** [comment]
@ZJZ0405 you are correct; YOLOv8's current structure does not natively support multi-label detection. Modifying the data loader and loss function to handle multiple classes per bounding box is necessary. If you have specific requirements, we encourage you to experiment and share your findings with the community.
